### PR TITLE
release: v1.45.0

### PR DESCRIPTION
**Pull request by bnjjj**
> **Note**
> **This particular PR must be true-merged to `main`.**

* This PR is only ready to review when it is marked as "Ready for Review".  It represents the merge to the `main` branch of an upcoming release (version number in the title).
* It will act as a staging branch until we are ready to finalize the release.
* We may cut any number of alpha and release candidate (RC) versions off this branch prior to formalizing it.
* This PR is **primarily a merge commit**, so reviewing every individual commit shown below is **not necessary** since those have been reviewed in their own PR.  However, things important to review on this PR **once it's marked "Ready for Review"**:
    - Does this PR target the right branch? (usually, `main`)
    - Are the appropriate **version bumps** and **release note edits** in the end of the commit list (or within the last few commits).  In other words, "Did the 'release prep' PR actually land on this branch?"
    - If those things look good, this PR is good to merge!

### CODE REVIEW

File: .changesets/feat_garypen_2002_subgraph_batching.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+```yaml
+batching:
+  enabled: true
+  mode: batch_http_link
+  subgraph:
+    # Disable batching on all subgraphs
+    all:
+      enabled: false
+    # Configure(over-ride) batching support per subgraph
```
Comment (shorgi) [inline .changesets/feat_garypen_2002_subgraph_batching.md:29]:
```suggestion
    # Configure (override) batching support per subgraph
```

File: .changesets/feat_garypen_2002_subgraph_batching.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    subgraphs:
+      subgraph_1:
+        enabled: true
+      subgraph_2:
+        enabled: true
+```
+
+Note: `all` may be over-ridden by `subgraphs`. This applies in general for all router subgraph configuration options.
```
Comment (shorgi) [inline .changesets/feat_garypen_2002_subgraph_batching.md:37]:
```suggestion
Note: `all` can be overridden by `subgraphs`. This applies in general for all router subgraph configuration options.
```

File: .changesets/fix_njm_p_681_pr_tweaks.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+### Performance tweaks of Apollo usage report field generation ([PR 4951](https://github.com/apollographql/router/pull/4951))
+
+Improves performance of the Apollo usage report signature/stats key and referenced field generation.
```
Comment (shorgi) [inline .changesets/fix_njm_p_681_pr_tweaks.md:3]:
```suggestion
### Performance improvements for Apollo usage report field generation ([PR 4951](https://github.com/apollographql/router/pull/4951))

The performance of generating Apollo usage report signatures, stats keys, and referenced fields has been improved.
```

File: .changesets/fix_watcher_raccoon_meat_crop.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+### Apply alias rewrites to arrays ([PR #TODO](https://github.com/apollographql/router/pull/4958))
+
+
+[#2489](https://github.com/apollographql/router/pull/2489) introduced automatic aliasing rules to support `@interfaceObject`.
+
+These rules now properly apply to lists.
```
Comment (shorgi) [inline .changesets/fix_watcher_raccoon_meat_crop.md:6]:
```suggestion
### Apply alias rewrites to arrays ([PR #TODO](https://github.com/apollographql/router/pull/4958))

The automatic aliasing rules introduced in [#2489](https://github.com/apollographql/router/pull/2489) to support `@interfaceObject` are now properly applied to lists.
```

File: .changesets/docs_press_finish_musket_reindeer.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+### [docs] Update caching with caveats ([PR #4872](https://github.com/apollographql/router/pull/4872))
+
+Add some words about what the operation cache intentions are
```
Comment (shorgi) [inline .changesets/docs_press_finish_musket_reindeer.md:3]:
```suggestion
### Documentation updates for caching and metrics instruments ([PR #4872](https://github.com/apollographql/router/pull/4872))

Router documentation has been updated for a couple topics:
- [Performance improvements vs. stability concerns](https://www.apollographql.com/docs/router/configuration/in-memory-caching#performance-improvements-vs-stability) when using the router's operation cache
- [Overview of standard and custom metrics instruments](https://www.apollographql.com/docs/router/configuration/telemetry/instrumentation/instruments) 
 
```

**Review by shorgi — COMMENTED** [review]
changelog review

File: .changesets/maint_240417_proto_update.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+### Updates the Apollo reporting protobuf to the latest version ([PR 4967](https://github.com/apollographql/router/pull/4967))
+
+Updates the protobuf file and related snapshots, and removes code that used the deprecated field.
```
[the remark this answers is not in the document]
Comment (abernix) [reply .changesets/maint_240417_proto_update.md:3]: No, we actually shouldn't even include this at all.  It's not really user-facing.  I'll delete the file.

**Review by o0Ignition0o — APPROVED** [review]

**Review by garypen — APPROVED** [review]

**Pull request merged**